Project-Aurora/Project-Aurora/Devices/OpenRGB/OpenRGBDevice.cs

```diff
@@ -33,291 +33,279 @@ public override bool Initialize()
 
             try
             {
+                // Get connection settings
                 var ip = Global.Configuration.VarRegistry.GetVariable<string>($"{DeviceName}_ip");
                 var port = Global.Configuration.VarRegistry.GetVariable<int>($"{DeviceName}_port");
-                var connectSleepTimeSeconds = Global.Configuration.VarRegistry.GetVariable<int>($"{DeviceName}_connect_sleep_time");
-                
-                bool openrgbRunning = false;
-                var processMonitor = RunningProcessMonitor.Instance;
-                void OnProcessMonitorOnRunningProcessesChanged(object o, RunningProcessChanged runningProcessChanged)
-                {
-                    if (processMonitor.IsProcessRunning("openrgb.exe"))
-                    {
-                        openrgbRunning = true;
-                    }
-                }
+                var remainingMillis = Global.Configuration.VarRegistry.GetVariable<int>($"{DeviceName}_connect_sleep_time") * 1000;
 
-                if (processMonitor.IsProcessRunning("openrgb.exe"))
-                {
-                    openrgbRunning = true;
-                }
-                processMonitor.RunningProcessesChanged += OnProcessMonitorOnRunningProcessesChanged;
-                
-                int remainingMillis = connectSleepTimeSeconds * 1000;
-                while (!openrgbRunning)
+                var openrgbRunning = () => Global.LightingStateManager.RunningProcessMonitor.IsProcessRunning("openrgb.exe");
+
+                // Search if process is running
+                while (!openrgbRunning())
                 {
                     Thread.Sleep(100);
                     remainingMillis -= 100;
                     if (remainingMillis <= 0)
                     {
-                        _openRgb = null;
-                        return false;
+                        throw new Exception("OpenRGB is not running...");
```

Comment:
can you also update DeviceDetails property? This was the message will be shown on UI, too.

Reply:
Yes!